### PR TITLE
Unit tests and cleanup

### PR DESCRIPTION
**Pull request by FrittenKeeZ**
Requires statamic/cms#5844 for unit tests to be fixed

### CODE REVIEW

File: src/Assets/AssetContainer.php [hunk 1]
```diff
@@ -74,7 +82,16 @@ public function save()
 
         $model->save();
 
-        $this->model($model->fresh());
+        $this->model($model->fresh())
```
Comment (ryanmitchell) [inline src/Assets/AssetContainer.php]: Feels like this should be calling fromModel to avoid repetition?

File: src/Assets/AssetContainerRepository.php [hunk 2]
```diff
@@ -43,7 +43,7 @@ public function make(string $handle = null): AssetContainerContract
 
     public function save(AssetContainerContract $container)
     {
-        $container->toModel()->save();
+        $container->save();
```
Comment (ryanmitchell) [inline src/Assets/AssetContainerRepository.php:46]: Did you test this change? Seems like something that was there on purpose.

File: database/migrations/2021_05_28_114212_create_collections_table.php [hunk 3]
```diff
@@ -17,22 +17,7 @@ public function up()
...
-            $table->string('template')->nullable();
-            $table->string('layout')->nullable();
-            $table->string('sort_dir')->nullable();
-            $table->string('sort_field')->nullable();
-            $table->string('mount')->nullable();
-            $table->json('taxonomies')->nullable();
-            $table->boolean('revisions')->default(false);
-            $table->json('inject')->nullable();
-            $table->json('structure')->nullable();
+            $table->json('settings')->nullable();
```
Comment (ryanmitchell) [inline database/migrations/2021_05_28_114212_create_collections_table.php:20]: But this will mean anyone upgrading won’t get the new code. I would revert this change.

File: src/Assets/AssetContainer.php (hunk 1 above)
Comment (FrittenKeeZ) [reply src/Assets/AssetContainer.php]: I know, but `fromModel` creates a new instance so we can't use that.
I could add a `fillFromModel` that both method could use.

File: src/Assets/AssetContainerRepository.php (hunk 2 above)
Comment (FrittenKeeZ) [reply src/Assets/AssetContainerRepository.php:46]: Yes, this is one of the reasons why the unit test failed - also AssetContainer is the one triggering the `AssetContainerSaved` event

File: src/Assets/AssetContainer.php (hunk 1 above)
Comment (ryanmitchell) [reply src/Assets/AssetContainer.php]: Yeah I think that would be good.

File: database/migrations/2021_05_28_114212_create_collections_table.php (hunk 3 above)
Comment (FrittenKeeZ) [reply database/migrations/2021_05_28_114212_create_collections_table.php:20]: I can definitely see where you're coming from, but as this is a "clean" PR with new functionality, I'd rather opt for existing users applying a composer patch we can provide in docs instead, as they will be the exception.

Comment (ryanmitchell) [reply database/migrations/2021_05_28_114212_create_collections_table.php:20]: They won’t. So many people use v0.2 of this. There is no benefit in squashing the migration that I can see. I didn’t do it originally intentionally.

Comment (FrittenKeeZ) [reply database/migrations/2021_05_28_114212_create_collections_table.php:20]: But v0.2 is only entries? Your PR provides migrations for everything else, so they'd have to use your branch from before February 15th 2022 for the change to be of any effect.

Comment (ryanmitchell) [reply database/migrations/2021_05_28_114212_create_collections_table.php:20]: Yeah valid point, in my head v2 also handled collections but I double checked and youre correct.

**Review by ryanmitchell — APPROVED** [review]
